src/appellate/appellate.js

```diff
@@ -133,9 +145,49 @@ AppellateDelegate.prototype.handleDocketDisplayPage = async function () {
     this.recap.uploadDocket(this.court, pacer_case_id, document.documentElement.innerHTML, 'APPELLATE_DOCKET', (ok) =>
       callback(ok)
     );
-
-    this.attachRecapLinksToEligibleDocs();
   } else {
     console.info(`RECAP: Not uploading docket. RECAP is disabled.`);
   }
 };
+
+AppellateDelegate.prototype.handleAttachmentPage = async function () {
+  if (history.state && history.state.uploaded) {
+    return;
+  }
+
+  let pacer_case_id = APPELLATE.getCaseIdFromSearchQuery(this.queryParameters) || APPELLATE.getCaseIdFromInputs();
+
+  // if the last step didn't find the caseId in the query parameter, It will check the storage
+  if (!pacer_case_id) {
+    const tabStorage = await getItemsFromStorage(this.tabId);
+    if (!tabStorage && !tabStorage.caseId) {
+      return;
+    }
+    pacer_case_id = tabStorage.caseId;
+  }
```

Comment:
Seeing this in a lot of places makes me think we should just have one utility function for this. Something like:

```
getCaseID: (queryParameters) => {
  // Do the search query part
  ...
  // Then do the from inputs part
  ...
  // Then do the from storage part
  ...
}
```

Reply:
You're right! A utility function will simplify things.

Thanks